src/System.Net.Mail/tests/Functional/SmtpClientTest.cs

```diff
@@ -312,5 +312,32 @@ public void TestMailDeliveryAsync()
 
             Assert.True(task.Wait(1000));
```

Comment:
I know this isn't part of this PR but GH showed it above: we've found that combining Wait and Async may lead to deadlocks or intermittent test failures. It's better to define the test `async` and await the task maybe using https://github.com/dotnet/corefx/blob/master/src/Common/tests/System/Threading/Tasks/TaskTimeoutExtensions.cs 

Also, it would be better to define a common timeout instead of hardcoding 1 second.

/cc: @stephentoub @karelz